app/src/main/java/org/literacyapp/authentication/AuthenticationHelper.java

```diff
@@ -0,0 +1,67 @@
+package org.literacyapp.authentication;
+
+import android.content.Context;
+import android.util.Log;
+
+import org.apache.commons.math3.analysis.function.StepFunction;
+import org.literacyapp.LiteracyApplication;
+import org.literacyapp.dao.AuthenticationEventDao;
+import org.literacyapp.dao.DaoSession;
+import org.literacyapp.model.Device;
+import org.literacyapp.model.Student;
+import org.literacyapp.model.analytics.AuthenticationEvent;
+import org.literacyapp.receiver.BootReceiver;
+import org.literacyapp.util.DeviceInfoHelper;
+import org.literacyapp.util.StudentUpdateHelper;
+
+import java.util.Calendar;
+import java.util.List;
+
+/**
+ * Created by sladomic on 30.12.16.
+ */
+
+public class AuthenticationHelper {
+    private static final double[] countSteps = { 0, 5, 10, 15, 20 };
+    private static final double[] exponentialDelayStepsInMinutes = { 30, 60, 120, 240, 360 };
```

Comment:
At the moment it would be 30 + 30 + 30 + 30 + 30 + 60 + 60 + 60 + 60 + 60 + 120 + 120 + 120 + 120 + 120 + 240 + 240 + 240 + 240 + 240 + 360 + 360 + 360 + 360 + 360 = 4050 minutes